### PR TITLE
[core] Introduce Source::setMinimumTileUpdateInterval API

### PR DESCRIPTION
**Pull request by pozdnyakov**
The `Source::setMinimumTileUpdateInterval(Duration)` method sets the minimum tile update interval, which is used to throttle the tile update network requests.

The corresponding `Source::getMinimumTileUpdateInterval()`  getter is added too.

Default minimum tile update interval value is `Duration::zero()`.

Tag https://github.com/mapbox/mapbox-gl-native-team/issues/335

### CODE REVIEW

**Comment by asheemmamoowala** [comment]
> @asheemmamoowala do you think that property would be useful for gl-js as well?

@pozdnyakov Is there any documentation around what this method is supposed to do? Tracking for parity at https://github.com/mapbox/mapbox-gl-js/issues/9603, but I don't see any use case for it in GL-JS as yet.

**Comment by pozdnyakov** [comment]
> @pozdnyakov Is there any documentation around what this method is supposed to do?

Just in this pr description, and the inline comments in `source.hpp`.. However, it's quite simple: it makes sure the update tile network request does not happen more often than the given duration (even if the server-provided expiration date supposes more frequent tile updates)

File: test/storage/online_file_source.test.cpp [hunk 1]
```diff
@@ -217,6 +217,25 @@ TEST(OnlineFileSource, TEST_REQUIRES_SERVER(RespectPriorExpires)) {
...
+
+    auto start = util::now();
+    Resource resource{Resource::Unknown, "http://127.0.0.1:3000/test"};
+    resource.priorExpires = start + std::chrono::duration_cast<Seconds>(Milliseconds(100));
+    resource.minimumUpdateInterval = Seconds(1);
+
+    std::unique_ptr<AsyncRequest> req = fs->request(resource, [&](Response) {
+        auto wait = util::now() - start;
+        EXPECT_GE(wait, resource.minimumUpdateInterval);
+        EXPECT_LT(wait, resource.minimumUpdateInterval + Milliseconds(10));
```
Comment (tmpsantos) [inline test/storage/online_file_source.test.cpp:232]: 1: [ RUN      ] OnlineFileSource.RespectMinimumUpdateInterval
1: ../test/storage/online_file_source.test.cpp:232: Failure
1: Expected: (wait) < (resource.minimumUpdateInterval + Milliseconds(10)), actual: 8-byte object <02-00 00-00 00-00 00-00> vs 8-byte object <80-60 33-3C 00-00 00-00>

This one here is flaking.

File: include/mbgl/style/source.hpp [hunk 2]
```diff
@@ -76,6 +79,14 @@ class Source : public mbgl::util::noncopyable {
     void setPrefetchZoomDelta(optional<uint8_t> delta) noexcept;
     optional<uint8_t> getPrefetchZoomDelta() const noexcept;
 
+    // If the given source supports loading tiles from a server,
+    // sets the minimum tile update interval, which is used to
+    // throttle the tile update network requests.
```
Comment (chloekraw) [inline include/mbgl/style/source.hpp:84]: nit: "suppress" or "cancel" might be a better word here, if we're preventing the update network requests from happening entirely. "throttle" sounds like the request is still made but we're slowing them down.

**Comment by chloekraw** [comment]
@pozdnyakov a few follow-up questions: 
- can `Source::setMinimumTileUpdateInterval(Duration)` be safely called at any time? 
- once a TTL override is set, can the same API call be used to change the override? For example, if a device were on low power mode, you might want to slow down a source's refresh rate even further.

---

cc @mapbox/maps-android @knov to cut tickets for platform bindings. For now, it's only a high priority on Android.

cc @asheemmamoowala

File: include/mbgl/style/source.hpp (hunk 2 above)
Comment (pozdnyakov) [reply include/mbgl/style/source.hpp:84]: In our requests are still made but less frequently, is "suppress"  still a better word?

Comment (chloekraw) [reply include/mbgl/style/source.hpp:84]: Hmm, so my thinking is, update network requests that are more frequent than the `minimumTileUpdateInterval` are suppressed (i.e., not made at all) -- is that accurate?

**Comment by pozdnyakov** [comment]
> - can `Source::setMinimumTileUpdateInterval(Duration)` be safely called at any time? 

yeah 

> - once a TTL override is set, can the same API call be used to change the override? For example, if a device were on low power mode, you might want to slow down a source's refresh rate even further.

absolutely, the resulting network request interval is always ` = max(TTL, minimuTileUpdateInterval /* set by the introduced API*/)`

File: include/mbgl/style/source.hpp (hunk 2 above)
Comment (pozdnyakov) [reply include/mbgl/style/source.hpp:84]: yeah, we can put it that way, I'll update this comment in https://github.com/mapbox/mapbox-gl-native/pull/16422

**Comment by chloekraw** [comment]
thanks for the clarification @pozdnyakov!

**Pull request merged**